buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-				)
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-				testExists(t,
-					filepath.Join(layersDir, "A"),
-				)
-			})
+						var actual string
+						t.Log("sets CNB_BUILDPACK_DIR")
+						actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-buildpack-dir-A-v1.clear"))
+						h.AssertEq(t, actual, descriptor.Dir)
 
-			it("should provide the platform dir", func() {
-				h.Mkfile(t, "some-data",
-					filepath.Join(platformDir, "env", "SOME_VAR"),
-				)
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-				testExists(t,
-					filepath.Join(appDir, "build-env-A-v1", "SOME_VAR"),
-				)
-			})
+						t.Log("sets CNB_PLATFORM_DIR")
+						actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-platform-dir-A-v1.clear"))
+						h.AssertEq(t, actual, platformDir)
 
-			it("should provide environment variables", func() {
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-				if s := cmp.Diff(h.Rdfile(t, filepath.Join(appDir, "build-info-A-v1")),
-					"TEST_ENV: Av1\n",
-				); s != "" {
-					t.Fatalf("Unexpected info:\n%s\n", s)
-				}
-			})
+						t.Log("sets CNB_BP_PLAN_PATH")
+						actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-bp-plan-path-A-v1.clear"))
+						if isUnset(actual) {
+							t.Fatal("Expected CNB_BP_PLAN_PATH to be set")
+						}
 
-			it("should set CNB_BUILDPACK_DIR", func() {
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-				if s := cmp.Diff(h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-buildpack-dir-A-v1")),
-					bpTOML.Dir,
-				); s != "" {
-					t.Fatalf("Unexpected CNB_BUILDPACK_DIR:\n%s\n", s)
-				}
-			})
+						if kind == buildpack.KindExtension {
+							t.Log("sets CNB_OUTPUT_DIR")
+							actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-output-dir-A-v1.clear"))
+							h.AssertEq(t, actual, filepath.Join(layersDir, "A"))
+							t.Log("does not set CNB_LAYERS_DIR")
+							actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-layers-dir-A-v1.clear"))
+							h.AssertEq(t, isUnset(actual), true)
+						} else {
+							t.Log("sets CNB_LAYERS_DIR")
+							actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-layers-dir-A-v1.clear"))
+							h.AssertEq(t, actual, filepath.Join(layersDir, "A"))
+							t.Log("does not set CNB_OUTPUT_DIR")
+							actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-output-dir-A-v1.clear"))
+							h.AssertEq(t, isUnset(actual), true)
+						}
+					})
+				})
 
-			it("should set CNB_LAYERS_DIR", func() {
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L187-L222